festim/generic_simulation.py

```diff
@@ -381,31 +381,41 @@ def initialise(self):
                                 f"SurfaceKinetics boundary condition must be defined on surface {q.surface} to export data with festim.AdsorbedHydrogen"
                             )
 
-        self.exports.initialise_derived_quantities(
-            self.mesh.dx, self.mesh.ds, self.materials
-        )
-
-        # needed to ensure that data is actually exported at TXTExport.times
-        # see issue 675
-        for export in self.exports:
-            if isinstance(export, festim.TXTExport) and export.times:
-                if not self.dt.milestones:
-                    self.dt.milestones = []
-                for time in export.times:
-                    if time not in self.dt.milestones:
-                        msg = "To ensure that TXTExport exports data at the desired times "
-                        msg += "TXTExport.times are added to milestones"
-                        warnings.warn(msg)
-                        self.dt.milestones.append(time)
-                self.dt.milestones.sort()
-
-            # set Soret to True for SurfaceFlux quantities
-            if isinstance(export, festim.DerivedQuantities):
-                for q in export:
+                    # set Soret to True for SurfaceFlux quantities
                     if isinstance(q, festim.SurfaceFlux):
                         q.soret = self.settings.soret
                         q.T = self.T.T
 
+            if isinstance(export, festim.TXTExport):
+                # pre-process data depending on the chemical potential flag, trap element type,
+                #  and material borders
+                project_to_DG = (
+                    self.settings.chemical_pot
+                    or self.settings.traps_element_type == "DG"
+                )
+                export.initialise_TXTExport(
+                    self.mesh.mesh,
+                    project_to_DG,
+                    self.materials,
+                )
```

Comment:
1. I removed the repeating loop `for export in self.exports` below. I don't think there is a needto iterate through it twice durinig initialisation.
2. I temporarly added new method `initialise_TXTExport` (former `filter_duplicates`) that's called once.
3. Considering [this ](https://github.com/festim-dev/FESTIM-workshop/blob/main/tasks/task13.ipynb) task of FESTIM WS, I assume that we should project fields to `DG` if `settings.traps_element_type = "DG"`. I added such a check with the `project_to_DG` variable, but it doesn't help and some information is lost due to filtering. For example, here is how the second figure from the task looks like: 
![image](https://github.com/user-attachments/assets/1cfee52a-8ed9-4f11-af86-1bb046aa57b0)
since there are borders, the output data near the boundary looks like this:
<details><summary>Details</summary>
<p>

4.646464646464646964e-01,5.000000000000542899e-01
4.747474747474748069e-01,5.000000000000532907e-01
4.848484848484848619e-01,5.000000000000531797e-01
**4.949494949494949725e-01,5.000000000000528466e-01
4.949494949494949725e-01,5.000000000000526246e-01**
5.050505050505050830e-01,5.000000000000521805e-01
5.151515151515152491e-01,0.000000000000000000e+00
5.252525252525253041e-01,0.000000000000000000e+00

</p>
</details> 

if there are no `material.borders`, then (fig. 4 from the task):
![image](https://github.com/user-attachments/assets/a1a9677d-3a67-4890-97d5-68a9f5ac4ed2)
and there are no duplicates in the output file.

Reply:
There's not much we can do about 3. I'm afraid. That is why I always recommend working with XDMF because there is no alteration of the data.

Reply:
However, what I showed is the result of filtering. We could filter data only for solute H or I don't know. If it's ok, then I'd maybe add a warning message